kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala

```diff
@@ -19,28 +19,136 @@ package com.datamountaineer.streamreactor.connect.influx.writers
+            case other => sys.error(s"$other schema is not supported")
+          }
+      }
+    }
     val batchPoints = BatchPoints
       .database(settings.database)
       .tag("async", "true")
       .retentionPolicy("default")
       .consistency(ConsistencyLevel.ALL)
       .build()
 
-    records.foreach(r => buildPoint(r, settings).map(batchPoints.point))
+    records.foreach(r => handleSinkRecord(r).map(batchPoints.point))
     batchPoints
   }
 
-  private def buildPoint(record: SinkRecord, settings: InfluxSettings): Option[Point] = {
+
+  private def buildPointFromMap(record: SinkRecord, settings: InfluxSettings): Option[Point] = {
+    require(record.value() != null && record.value().getClass == classOf[java.util.HashMap[_, _]],
+      "The SinkRecord payload should be of type java.util.Map[String, Any]")
+
+    val map = record.value().asInstanceOf[java.util.Map[String, Any]]
+
+    val extractor = settings.fieldsExtractorMap(record.topic())
+    val timestamp = extractor.timestampField.map { field =>
+      map.getOrElse(field,
+        throw new ConfigException(s"$field has not been found on the record on topic ${record.topic()} partition:${record.kafkaPartition()} and offset:${record.kafkaOffset()}"))
+      match {
+        case b: Byte => b.toLong
+        case s: Short => s.toLong
+        case i: Int => i.toLong
+        case l: Long => l
+        case _ => throw new ConfigException(s"$field is not a valid field for the timestamp")
+      }
+    }.getOrElse(System.currentTimeMillis())
+
+    val convertedMap = convertSchemalessJson(record, extractor.fieldsAliasMap, extractor.ignoredFields, false, extractor.includeAllFields)
+
+    fromMapToPoint(convertedMap, timestamp, settings, record)
+  }
+
+  private def fromMapToPoint(map: java.util.Map[String, Any],
+                             timestamp: Long,
+                             settings: InfluxSettings,
+                             record: SinkRecord): Option[Point] = {
+    if (map.nonEmpty) {
+      val builder = Point.measurement(settings.topicToMeasurementMap.getOrElse(record.topic(),
+        throw new ConfigException(s"No matching measurement for topic ${record.topic}")))
+        .time(timestamp, TimeUnit.MILLISECONDS)
+
+      map.foreach {
+        case (field, value: Long) => builder.addField(field, value)
+        case (field, value: Int) => builder.addField(field, value)
+        case (field, value: BigInt) => builder.addField(field, value)
+        case (field, value: Byte) => builder.addField(field, value)
+        case (field, value: Short) => builder.addField(field, value)
+        case (field, value: Double) => builder.addField(field, value)
+        case (field, value: Float) => builder.addField(field, value)
+        case (field, value: Boolean) => builder.addField(field, value)
+        case (field, value: java.math.BigDecimal) => builder.addField(field, value)
+        case (field, value: String) => builder.addField(field, value)
```

Comment:
What if the nested has fields already processed? basically you do an ovewrite? For now we don't support them